### PR TITLE
statistics: add mutex for Handle.globalMap and Handle.feedback

### LINKED ISSUE #29181: Handle has potential data race
## Enhancement

`updateStatsWorker` periodically calls `(*Handle).DumpStatsDeltaToKV` and `(*Handle).UpdateStatsByLocalFeedback` and the both functions calls `(*Handle).sweepList`. In another word, `(*Handle).sweepList` may be concurrently called by different callers. `(*Handle).sweepList` calls `(*Handle).merge`, which writes `Handle.globalMap` and `Handle.feedback` without acquiring lock. Hence data race may occur. Maybe we should prevent concurrently writing `Handle.globalMap` and `Handle.feedback`.

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #29181

Problem Summary:
There exists potential data race when accessing `Handle.globalMap`/`Handle.feedback` concurrently.

### What is changed and how it works?
Add mutex for `Handle.globalMap` and `Handle.feedback` respectively. Each time we want to update `Handle.globalMap`/`Handle.feedback`, lock the mutex first.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -159,6 +165,7 @@ func (h *Handle) execRestrictedSQLWithSnapshot(ctx context.Context, sql string,
 
 // Clear the statsCache, only for test.
 func (h *Handle) Clear() {
+	// TODO: Here h.mu seems to protect all the fields of Handle. Is is reasonable?
```
Comment (winoros) [inline statistics/handle/handle.go:168]: After this pr, the `h.mu` doesn't protect all the fields of the Handle, right?

**Review by winoros — COMMENTED** [review]
Can we build a test using the failpoint?

File: statistics/handle/handle.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply statistics/handle/handle.go:168]: IMHO, `h.mu` never protects all the fields of the Handle. It is only responsible for the fields in its own struct(`ctx`, `rateMap`, `pid2tid` and `schemaVersion`). `h.statsCache` has its own mutex and is not protected by `h.mu`.

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**